feat(core): Remove `storeMetadata` and `getSize` from binary data manager interface (no-changelog)

ivov opened the pull request:
Depends on: #7164 | Story: [PAY-838](https://linear.app/n8n/issue/PAY-838/introduce-object-store-service-for-binary-data)

This PR removes `storeMetadata` and `getSize` from the binary data manager interface, as these are specific to filesystem mode. Also this disambiguates identifiers:

```
binaryDataId
filesystem:289b4aac51e-dac6-4167-b793-6d5c415e2b47 {mode}:{fileId}

fileId - FS
289b4aac51e-dac6-4167-b793-6d5c415e2b47 {executionId}{uuid}

fileId - S3
/workflows/{workflowId}/executions/{executionId}/binary_data/b4aac51e-dac6-4167-b793-6d5c415e2b47
```

Note: The object store changes originally in this PR were extracted out into the final PR.

netroy committed:
refactor(core): Simplify executions and binary data pruning

ivov committed:
Merge branch 'master' into pay-771-implement-soft-deletion

ivov committed:
Merge branch 'master' into pay-771-implement-soft-deletion

ivov committed:
Remove test code from service

ivov committed:
Use time constants

ivov committed:
Improve naming

ivov committed:
Use native typeorm soft deletion

ivov committed:
Improve naming

ivov committed:
Make batch size class field

ivov committed:
Cleanup

ivov committed:
Remove unused method

ivov committed:
Cleanup

ivov committed:
Filter out soft-deleted executions

ivov committed:
Merge branch 'master' into pay-771-implement-soft-deletion

ivov committed:
Add tests

ivov committed:
Improve test

ivov committed:
Soft-delete in single pass

ivov committed:
Restore value

ivov committed:
Restore from debug value

ivov committed:
Add clarifying comments

ivov committed:
Update tests

ivov committed:
Add `typedi` to core

ivov committed:
Update package.json

ivov committed:
Turn `BinaryDataManager` into service

ivov committed:
Add `object` to binary data schema options

ivov committed:
Use service throughout core and cli

ivov committed:
Update tests

ivov committed:
Reduce diff

ivov committed:
Update default

ivov committed:
Add docline

ivov committed:
Remove redundant output types

ivov committed:
Remove unused type

ivov committed:
Rename to submanagers to client

ivov committed:
Rename `BinaryDataFileSystem` to `FileSystemClient`

ivov committed:
Clean up types

ivov committed:
Move `binaryToBuffer` to manager

ivov committed:
Rename `BinaryDataManager` to `BinaryDataService`

ivov committed:
Unify storage path naming

ivov committed:
Set binary data service in nodes-base test helper

ivov committed:
Rename manager in nodes-base

ivov committed:
More renamings

ivov committed:
Generalize interface

ivov committed:
Continue generalizing

ivov committed:
Fix test

ivov committed:
Better naming

ivov committed:
More renamings

ivov committed:
Minor cleanup

ivov committed:
More minor cleanup

ivov committed:
Add clarifying comment

ivov committed:
Speed up pruning if high volume

ivov committed:
Merge branch 'master' into pay-771-implement-soft-deletion

ivov committed:
Remove call from hook

ivov committed:
Merge parent branch

ivov committed:
Refix conflict

ivov committed:
Add formats to schema

ivov committed:
Add `ensureStringArray` util

ivov committed:
Adjust types based on parsing

ivov committed:
Add type guard

ivov committed:
Clean up types in `init()`

ivov committed:
Fix test

ivov committed:
Remove unneeded modifier

ivov committed:
Cleanup

ivov committed:
Fix node tests

ivov committed:
Remove unneeded line

ivov committed:
Rename constant

ivov committed:
Remove comment

ivov committed:
Fix tests

ivov committed:
Cleanup

ivov committed:
Rename error

ivov committed:
Improve error message

ivov committed:
Better error

ivov committed:
Better message

ivov committed:
Make execution ID non-nullable

ivov committed:
Merge master

ivov committed:
Readability improvements

ivov committed:
Adjust types, followup to 67e163f

ivov committed:
Merge parent branch

ivov committed:
Fix lint

ivov committed:
Fix lint

ivov committed:
Cleanup

ivov committed:
Install `aws4` and `@types/aws4`

ivov committed:
Add to config schema

ivov committed:
Set up object store service

ivov committed:
Set up temp dev code

ivov committed:
Comment out cache keys

ivov committed:
Revert "Comment out cache keys"

This reverts commit 96ea9eab6de78a9b8e494beacdcf413161373e17.

ivov committed:
Rename dir to `TempBinaryData`

ivov committed:
Rename back to `BinaryData`

csuermann commented on packages/cli/src/config/schema.ts:
```diff
@@ -908,6 +908,40 @@ export const schema = {
+					doc: 'Region of the n8n bucket in S3-compatible external storage',
+				},
+			},
+			credentials: {
```
Just food for thought: It might be useful to have the ability to manage S3 credentials in the same way as regular n8n credentials.

ivov replied:
Thanks! Consolidating [here](https://linear.app/n8n/issue/PAY-769/introduce-object-store-manager-for-binary-data#comment-db61b74b).

ivov committed:
Merge master

ivov committed:
Clear timers on shutdown

ivov committed:
Set timers only on main instance

ivov committed:
Also for clearing timers

ivov committed:
Merge parent branch

ivov committed:
Rename back to `localStoragePath`

ivov committed:
Remove unused arg

ivov committed:
Fix lint

ivov committed:
Merge parent branch

ivov committed:
Make as enterprise edition

ivov committed:
Add logging, refactor for readability

ivov committed:
Ensure hard-deletion select includes soft-deleted rows

ivov committed:
Switch `info` to `debug`

ivov committed:
Fix tests

ivov committed:
Remove redundant checks for `deletedAt` being `NULL`

The column `deletedAt` is marked `@DeleteDateColumn()` so all reads
from the repository automatically add a `WHERE` clause checking
that the column `IS NULL`.

ivov committed:
Fix lint

ivov committed:
Fix last test

ivov committed:
More setup

ivov committed:
Add sample path structure

ivov committed:
Set up some errors

ivov committed:
Remove unused

ivov committed:
Cleanup

ivov committed:
Fix `HEAD` request

ivov committed:
More missing loggers in tests

ivov committed:
Add logger to even more test

ivov committed:
Fix upload

ivov committed:
Refactor logging for tests

ivov committed:
Consolidate `getStream` and `getBuffer`

ivov committed:
Cleanup

ivov committed:
Support single deletion

ivov committed:
Cleanup

ivov committed:
More cleanup

ivov committed:
Merge parent branch

ivov committed:
Merge parent branch

ivov committed:
Remove sample image

ivov committed:
Add listing

ivov committed:
Implement `deleteMany`

ivov committed:
Rename arg

ivov committed:
Major refactor to keep reducing interface

ivov committed:
Merge master, fix conflicts

ivov committed:
Fix misresolved conflict

ivov committed:
Remove excess check on init

ivov committed:
Merge parent branch

ivov committed:
Fix misresolved conflicts

ivov committed:
Remove commented out bits

ivov committed:
Remove comment

ivov committed:
Consistent naming

ivov committed:
Reorder

ivov committed:
Better error message

ivov committed:
Remove `ObjectStore` from this PR

ivov committed:
Remove schema change from this PR

ivov committed:
Update deps

ivov committed:
Remove missing dep

ivov committed:
Remove reference

ivov committed:
Cleanup

ivov committed:
Remove `ObjectStore.manager.ts` stub

ivov committed:
Remove `ObjectStore.manager.ts` stub

ivov committed:
Merge parent branch

ivov committed:
Revert changes to lockfile

ivov committed:
Cleanup

ivov committed:
Cleanup

ivov committed:
Remove unneeded comment

ivov committed:
Better naming

ivov committed:
Cleanup

ivov committed:
Move `ensureDirExists` to utils

ivov committed:
Consolidate errors

ivov committed:
Improve typings

ivov committed:
Restore error

ivov committed:
Revert rename to `BinaryDataManager`

ivov committed:
Merge parent branch

ivov committed:
More accurate naming

ivov committed:
Missing renaming in test

ivov committed:
Merge parent branch

ivov committed:
Tighten typings

ivov committed:
Rename constant

ivov committed:
Merge master

ivov committed:
Restore `LogCatch`

`LogCatch` was merged into master at a different file: `/packages/core/binaryData/index.ts`, which is now at `BinaryData.service.ts`

ivov committed:
Add file not found error to `getSize`

ivov committed:
Add `mode` to `InvalidBinaryDataManagerError`

ivov committed:
Merge parent branch

ivov committed:
Resolve more conflicts

krynble reviewed (dismissed):
LGTM.

I fear this part of the code is untested, but it's something we should be tackling maybe as a separate PR, like tech debt. Especially the service should be tested once we settle on how it should look like so we avoid breaking it.

ivov committed:
Merge master, fix conflicts

ivov committed:
Fix misresolved conflict

ivov committed:
Minor renaming

janober commented:
Got released with [`n8n@1.9.0`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.9.0)
